### PR TITLE
Fix: slack repeated callback event message for attached files

### LINKED ISSUE #4591: Message with Attachment triggers an unnecessary message if Slack is connected
**To Reproduce:**

- Connect Slack integration to the account
- Create a message with an attachment

**Dashboard View:** 

Screenshot 2022-05-02 at 8 05 08 PM

**Slack Thread View:**

Screenshot 2022-05-02 at 8 06 03 PM

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Fixing "Attached File!" message from the file attachment message from slack.

Fixes #4591 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Tested on staging and checked all the tests are passing well.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/integrations/slack/incoming_message_builder.rb [hunk 1]
```diff
@@ -135,4 +135,8 @@ def file_type(attachment)
       :file
     end
   end
+
+  def attached_file_message?
```
Comment (sojan-official) [inline lib/integrations/slack/incoming_message_builder.rb:139]: can we add a link/documentation on when this event occurs / why?

lets also add a test case for this condition.

Comment (tejaswinichile) [reply lib/integrations/slack/incoming_message_builder.rb:139]: @sojan-official 
This is what we are adding from our side no slack changes, we add it as [initial comment for the attached file](https://github.com/chatwoot/chatwoot/blob/develop/lib/integrations/slack/send_on_slack_service.rb#L69) so when we fixed receiving the message from slack other than the supported message [here](https://github.com/chatwoot/chatwoot/pull/4494/commits/07a1c0da1e522d76e37b5f0cecdb4613389ab9b6) it is saving the 'Attached file!' message in CW dashboard when we get postback message for the same.

So the flow is:
1. You send a file from CW to the customer with the attached file which adds an 'Attached File!' message.
2. Which sends to slack and sends send it back to the CW with callback
3. Before, We had checked for the same; if it supported the message, but now we have one more condition which passes for the 'Attached File!' message.
4. Now I am adding this check-in 'Incoming message builder' if the message is 'Attached File!' ignore it and prevent it from creating in the CW.

Adding the test cases for the same.

Comment (sojan-official) [reply lib/integrations/slack/incoming_message_builder.rb:139]: @tejaswinichile makes sense.  can we add this as a comment in the code please :D. I am pretty sure someone will get confused with this condition down the lane when they look at the code.

**Comment by tejaswinichile** [comment]
@sojan-official I have added the comment, please have a look.

**Review by sojan-official — APPROVED** [review]

**Pull request merged**